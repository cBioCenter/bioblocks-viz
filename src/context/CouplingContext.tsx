import * as React from 'react';

import {
  ResidueContext,
  ResidueContextProvider,
  SecondaryStructureContext,
  SecondaryStructureContextProvider,
} from '~chell-viz~/context';
import { CouplingContainer } from '~chell-viz~/data';

export const initialCouplingContext = {
  couplingScores: new CouplingContainer(),
};

export type ICouplingContext = typeof initialCouplingContext;

export const CouplingContext = React.createContext(initialCouplingContext);

/**
 * Shorthand for passing contexts relevant for Coupling Scores - Primarily interaction with residues and secondary structures.
 *
 * @export
 * @extends {React.Component<any, ICouplingContext>}
 */
export class CouplingContextProvider extends React.Component<any, ICouplingContext> {
  constructor(props: any) {
    super(props);
    this.state = {
      couplingScores: new CouplingContainer(),
    };
  }

  public render() {
    return (
      <SecondaryStructureContextProvider>
        <ResidueContextProvider>
          <CouplingContext.Provider value={this.state}>{this.props.children}</CouplingContext.Provider>
        </ResidueContextProvider>
      </SecondaryStructureContextProvider>
    );
  }
}

export const withCouplingContext = (Component: any) => (props: any) => (
  <CouplingContext.Consumer>
    {couplingContext => (
      <SecondaryStructureContext.Consumer>
        {secStructContext => (
          <ResidueContext.Consumer>
            {residueContext => (
              <Component
                {...props}
                residueContext={{ ...residueContext }}
                secondaryStructureContext={{ ...secStructContext }}
                couplingContext={{ ...couplingContext }}
              />
            )}
          </ResidueContext.Consumer>
        )}
      </SecondaryStructureContext.Consumer>
    )}
  </CouplingContext.Consumer>
);
