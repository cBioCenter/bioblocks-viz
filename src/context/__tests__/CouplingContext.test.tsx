import { shallow } from 'enzyme';
import * as React from 'react';

import { CouplingContextClass } from '../CouplingContext';

describe('CouplingContext', () => {
  it('Should match the default snapshot.', () => {
    const wrapper = shallow(<CouplingContextClass />);
    expect(wrapper).toMatchSnapshot();
  });
});
