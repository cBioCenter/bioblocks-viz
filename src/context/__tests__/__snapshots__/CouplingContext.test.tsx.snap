// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouplingContext Should match the default snapshot. 1`] = `
<SecondaryStructureContext>
  <ResidueContext>
    <ContextProvider
      value={
        Object {
          "couplingScores": CouplingContainer {
            "colCounter": 0,
            "contacts": Array [],
            "getCouplingScore": [Function],
            "includes": [Function],
            "rowCounter": 0,
            "totalStoredContacts": 0,
          },
        }
      }
    />
  </ResidueContext>
</SecondaryStructureContext>
`;
